sklearn/utils/validation.py

```diff
@@ -890,9 +898,14 @@ def check_is_fitted(estimator, msg=None):
     if not hasattr(estimator, 'fit'):
         raise TypeError("%s is not an estimator instance." % (estimator))
 
-    attrs = [v for v in vars(estimator)
-             if (v.endswith("_") or v.startswith("_"))
-             and not v.startswith("__")]
+    if attributes is not None:
+        if not isinstance(attributes, (list, tuple)):
+            attributes = [attributes]
+        attrs = all([hasattr(estimator, attr) for attr in attributes])
+    else:
+        attrs = [v for v in vars(estimator)
+                 if (v.endswith("_") or v.startswith("_"))
```

Comment:
From what I see in the issue, we want to get away from `startwith("_")`.

Reply:
+1 I came here to say so :)